src/users/users.controller.ts

```diff
@@ -123,10 +125,28 @@ export class UsersController {
       },
     },
   })
-  async getUserById(@Param('id') id: number): Promise<UserDto> {
+  async getFullUser(@Param('query') query: string): Promise<FullUserDto> {
+    try {
+      this.logger.debug(`Find a full user by email/id: ${query}`);
+      return new FullUserDto(await this.usersService.findUser(query));
+    } catch (err) {
+      throw new HttpException(err.message, err.status);
+    }
+  }
+
+  @Get('/search/:query')
+  @ApiOperation({
+    description: SWAGGER_DESC_FIND_USERS,
+  })
+  @ApiOkResponse({
+    type: UserDto,
+    description: 'Returns all array with user objects',
```

Comment:
The wording is unclear
```suggestion
    description: 'Returns the list of users',
```